eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1306 @@
+                             void                     *userdata,
+                             mustache_token_section_t *token,
+                             GVariant                 *variant)
+{
+  /* Need to render each sub-template from here */
+  if (g_variant_get_boolean (variant))
+    return mustache_render (api, userdata, token->section);
+
+  /* Nothing to do */
+  return 1;
+}
+
+static uintmax_t
+_renderer_str_sect_from_ht (mustache_api_t           *api,
+                            void                     *userdata,
+                            mustache_token_section_t *token,
+                            GVariant                 *variant)
+{
+  RendererMustacheData *data = userdata;
+  uintmax_t rv = 0;
+  const char *last_section_variable = data->section_variable;
+
+  data->section_variable = g_variant_get_string (variant, NULL);
+
+  /* Need to render each sub-template from here */
+  rv = mustache_render (api, userdata, token->section);
+
+  data->section_variable = last_section_variable;
+
+  return rv;
+}
+
+static uintmax_t
+_renderer_section_from_ht_variant (GVariant                 *variant,
+                                   mustache_api_t           *api,
+                                   void                     *userdata,
+                                   mustache_token_section_t *token)
+{
+  g_autofree char *msg = NULL;
+
+  if (g_variant_is_of_type (variant, G_VARIANT_TYPE_STRING_ARRAY))
+    return _renderer_strv_sect_from_ht (api, userdata, token, variant);
+  else if (g_variant_is_of_type (variant, G_VARIANT_TYPE_BOOLEAN))
+    return _renderer_bool_sect_from_ht (api, userdata, token, variant);
+  else if (g_variant_is_of_type (variant, G_VARIANT_TYPE_STRING))
+    return _renderer_str_sect_from_ht (api, userdata, token, variant);
+
+  msg = g_strdup_printf ("No handler for section type %s on token %s",
+                         g_variant_get_type_string (variant),
+                         token->name);
+  (*api->error) (api, userdata, __LINE__, msg);
+  return 1;
+}
+
+static uintmax_t
+_renderer_sect_from_ht (mustache_api_t           *api,
+                        void                     *userdata,
+                        mustache_token_section_t *token)
+{
+  RendererMustacheData *data = userdata;
+  g_autoptr(GVariant) value = g_variant_dict_lookup_value (data->variables,
+                                                           token->name,
+                                                           NULL);
+
+  if (value == NULL)
+    {
+      g_autofree char *msg = g_strdup_printf ("No such section %s", token->name);
+      (*api->error) (api, data, __LINE__, msg);
+      return 0;
+    }
+
+  if (!_renderer_section_from_ht_variant (value, api, userdata, token))
+    return 0;
+
+  return 1;
+}
+
+static void
+_renderer_set_error (G_GNUC_UNUSED mustache_api_t *api,
+                     void                         *userdata,
+                     G_GNUC_UNUSED uintmax_t       lineno,
```

Comment:
Might be good to include the `lineno` in the error message, for help with debugging.